# HG changeset patch
# User Rik <rik@octave.org>
# Date 1350416263 25200
#      Tue Oct 16 12:37:43 2012 -0700
# Node ID 39fd2ad26fb7132988456d14a5f74d9bb2df9b95
# Parent  203c78063bff3556b7c20a05416691fb5664ec29
Fix loading of empty structures from .mat files (bug #37087).

* ls-mat5.cc (read_mat5_binary_element): Size Octave map to the dimensions
read from the file, rather than assuming 1x1.

diff --git a/libinterp/interp-core/ls-mat5.cc b/libinterp/interp-core/ls-mat5.cc
--- a/libinterp/interp-core/ls-mat5.cc
+++ b/libinterp/interp-core/ls-mat5.cc
@@ -1139,17 +1139,17 @@ read_mat5_binary_element (std::istream& 
 
           name[len] = '\0';
           classname = name;
         }
       }
       // Fall-through
     case MAT_FILE_STRUCT_CLASS:
       {
-        Octave_map m (dim_vector (1, 1));
+        Octave_map m (dims);
         int32_t fn_type;
         int32_t fn_len;
         int32_t field_name_length;
 
         // field name length subelement -- actually the maximum length
         // of a field name.  The Matlab docs promise this will always
         // be 32.  We read and use the actual value, on the theory
         // that eventually someone will recognize that's a waste of
